src/journal/journalctl.c

```diff
@@ -1203,6 +1203,15 @@ static int get_boots(
                                 break;
                         }
                 } else {
+                        i = 0;
+                        LIST_FOREACH(boot_list, id, head) {
+                                if (sd_id128_equal(id->id, current->id)) {
+                                        /* boot id already stored, something wrong with the journal files */
+                                        /* exiting as otherwise this problem would cause forever loop */
+                                        goto finish;
+                                }
+                                i++;
```

Comment:
Please don't add such cleanups to your patches as extra commits but squash them together. In your case, this should be only one commit, not 3.